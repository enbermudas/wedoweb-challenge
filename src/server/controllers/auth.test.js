import db from '../config/sequelize';
import AuthController from './auth';

beforeAll(async () => await db.sequelize.sync({ alter: true, force: false }));

afterAll(async () => {
  await db.sequelize.dropAllSchemas({});
  db.sequelize.close();
});

it('Should sign up a new a user.', async () => {
  const mockRequest = {
    body: {
      username: 'testuser',
      email: 'testemail@email.com',
      password: '12345678'
    }
  };

  const mockResponse = {
    send: jest.fn(() => mockResponse),
    status: jest.fn(() => mockResponse)
  };

  await AuthController.signup(mockRequest, mockResponse);

  expect(mockResponse.status).toBeCalledWith(201);
  expect(mockResponse.send.mock.calls[0][0]).toMatchSnapshot();
});

it('Should fail if the unique constraint is violated.', async () => {
  const mockRequest = {
    body: {
      username: 'testuser',
      email: 'testemail@email.com',
      password: '12345678'
    }
  };

  const mockResponse = {
    send: jest.fn(() => mockResponse),
    status: jest.fn(() => mockResponse)
  };

  await AuthController.signup(mockRequest, mockResponse);

  expect(mockResponse.status).toBeCalledWith(500);
  expect(mockResponse.send.mock.calls[0][0]).toMatchSnapshot();
});

it('Should fail if there a missing fields.', async () => {
  const mockRequest = {
    body: {
      username: 'testuser',
      email: 'testemail@email.com'
    }
  };

  const mockResponse = {
    send: jest.fn(() => mockResponse),
    status: jest.fn(() => mockResponse)
  };

  await AuthController.signup(mockRequest, mockResponse);

  expect(mockResponse.status).toBeCalledWith(400);
  expect(mockResponse.send.mock.calls[0][0]).toMatchSnapshot();
});
