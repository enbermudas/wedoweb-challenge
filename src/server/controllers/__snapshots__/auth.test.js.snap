// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should fail if the unique constraint is violated. 1`] = `
Object {
  "message": Array [
    ValidationErrorItem {
      "instance": Object {
        "createdAt": 2021-01-16T19:46:43.570Z,
        "email": "testemail@email.com",
        "id": null,
        "updatedAt": 2021-01-16T19:46:43.570Z,
        "username": "testuser",
      },
      "message": "users.username must be unique",
      "origin": "DB",
      "path": "users.username",
      "type": "unique violation",
      "validatorArgs": Array [],
      "validatorKey": "not_unique",
      "validatorName": null,
      "value": "testuser",
    },
  ],
}
`;

exports[`Should fail if there a missing fields. 1`] = `
Object {
  "errors": Array [
    Object {
      "field": "password",
      "message": "The password is a required field.",
    },
  ],
  "message": "Validation failed.",
}
`;

exports[`Should sign up a new a user. 1`] = `
Object {
  "data": Object {
    "createdAt": 2021-01-16T19:46:43.480Z,
    "email": "testemail@email.com",
    "id": 1,
    "updatedAt": 2021-01-16T19:46:43.480Z,
    "username": "testuser",
  },
  "message": "You have successfully signed up.",
}
`;
